### PR TITLE
feat(nest): add e2e for nest

### PR DESCRIPTION
**Pull request by ndcunningham**
- Refactor the Nest default application to be similar to the other frameworks (express, fastify, koa)
- Refactor node `e2e-project` to include nest

### CODE REVIEW

File: packages/node/src/generators/application/application.ts [hunk 1]
```diff
@@ -400,6 +406,7 @@ export async function applicationGenerator(tree: Tree, schema: Schema) {
       name: options.rootProject ? 'e2e' : `${options.name}-e2e`,
       project: options.name,
       port: options.port,
+      isNest: options.isNest,
```
Comment (ndcunningham) [inline packages/node/src/generators/application/application.ts:409]: necessary, since `options.framework` is not passed back to the node generator.

**Comment by jaysoo** [comment]
Make sure the commits are squashed, and the PR title is correctly matching the commit message.

**Comment by nartc** [comment]
#15484 has been merged, can you rebase real quick so we don't have changes from 15484 in this one?

File: e2e/node/src/node-server.test.ts [hunk 2]
```diff
@@ -76,18 +77,27 @@ describe('Node Applications + webpack', () => {
...
+    runCLI(
+      `generate @nrwl/node:app ${nestApp} --framework=nest --bundler=webpack --no-interactive`
+    );
 
     // Use esbuild by default
     checkFilesDoNotExist(`apps/${expressApp}/webpack.config.js`);
     checkFilesDoNotExist(`apps/${fastifyApp}/webpack.config.js`);
     checkFilesDoNotExist(`apps/${koaApp}/webpack.config.js`);
 
+    // Useas only webpack
```
Comment (nartc) [inline e2e/node/src/node-server.test.ts]:
```suggestion
    // Uses only webpack
```

**Review by nartc — APPROVED** [review]

**Review by jaysoo — APPROVED** [review]

**Pull request merged**